chiselFrontend/src/main/scala/chisel3/internal/Error.scala

```diff
@@ -28,8 +28,16 @@ private[chisel3] class ErrorLog {
     println(new Info("[%2.3f] %s".format(elapsedTime/1e3, m), None))  // scalastyle:ignore regex
 
   /** Log a deprecation warning message */
-  def deprecated(m: => String): Unit = {
-    val thisEntry = (m, getUserLineNumber)
+  def deprecated(m: => String, location: Option[String]): Unit = {
+    val sourceLoc = location match {
+      case Some(loc) => loc
+      case None => getUserLineNumber match {
+        case Some(elt: StackTraceElement) => s"${elt.getFileName}:${elt.getLineNumber}"
+        case None => "(unknown)"
+      }
+    }
```

Comment:
I think this could better be expressed as:
```scala
val sourceLoc = location.orElse(getUserLineNumber)
                        .map(elt => s"${elt.getFileName}:${elt.getLineNumber})
                        .getOrElse("(unknown)")
```
but that's just like, my opinion man.

Reply:
Couldn't get that to work (`value getFileName is not a member of java.io.Serializable`), I also think the nested match makes the logic more clear.